From: =?UTF-8?q?Uwe=20Kleine-K=C3=B6nig?= <u.kleine-koenig@pengutronix.de>
Date: Tue, 3 Sep 2013 21:12:46 +0200
Subject: [PATCH] Don't disable udp and tcp when --disable-agentx-dom-sock-only
 is given

The configure magic without this patch treats all

	--enable-agentx-dom-sock-only
	--enable-agentx-dom-sock-only=no
	--disable-agentx-dom-sock-only

the same; i.e. disables the udp and tcp transports. With this patch only the
first option does have this effect which matches the usual and expected
semantic.
---
 configure.d/config_project_with_enable | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/configure.d/config_project_with_enable b/configure.d/config_project_with_enable
index cdf56deb69dc..37d6ed1d6984 100644
--- a/configure.d/config_project_with_enable
+++ b/configure.d/config_project_with_enable
@@ -771,9 +771,11 @@ NETSNMP_ARG_ENABLE(local-smux,
     fi])
 
 NETSNMP_ARG_ENABLE(agentx-dom-sock-only,
-[  --enable-agentx-dom-sock-only   Disable UDP/TCP transports for agentx.],
-AC_DEFINE(NETSNMP_AGENTX_DOM_SOCK_ONLY, 1,
-    [define if agentx transport is to use domain sockets only]))
+[  --enable-agentx-dom-sock-only   Disable UDP/TCP transports for agentx.])
+if test "x$enable_agentx_dom_sock_only" = "xyes"; then
+  AC_DEFINE(NETSNMP_AGENTX_DOM_SOCK_ONLY, 1,
+    [define if agentx transport is to use domain sockets only])
+fi
 
 NETSNMP_ARG_ENABLE(snmptrapd-subagent,
 [  --disable-snmptrapd-subagent    Disable agentx subagent code in snmptrapd.])
